### PR TITLE
Manage tags

### LINKED ISSUE #1: Manage tags and tag groups
It should be possible to manage tag groups and tags.

Both entities are can be queried and updated via GraphQL. This different in comparison to the other model files.

### PR DESCRIPTION
**Pull request by harti2006**
Fixes #1 

+ contains a lot of refactoring

### CODE REVIEW

File: leanix_admin/action.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            self.do_perform()
+            print('done.')
+        else:
+            print('skipped.')
+
+    def do_perform(self):
+        pass
+
+
+class BackupAction(ConfirmableAction):
```
Comment (sebastianpoeplau) [inline leanix_admin/action.py]: Java style! 😄

File: leanix_admin/auth.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        response = self.http.get(self.mtm_base_url + '/workspaces/' + workspace_id)
+        response.raise_for_status()
+        workspace_name = response.json()['data']['name']
+        print('Logged in to workspace:', workspace_name)
+
+    def _parse_jwt_payload(self):
+        jwt = self.http.auth.obtain_access_token()
+        payload_part = jwt.split('.')[1]
+        # fix missing padding for this base64 encoded string.
+        # If number of bytes is not dividable by 4, append '=' until it is.
```
Comment (sebastianpoeplau) [inline leanix_admin/auth.py]: Minor typo: divisible

Comment (sebastianpoeplau) [reply leanix_admin/auth.py]: Actually, dividable seems to be a word too... Never heard it 😢

File: leanix_admin/cli.py [hunk 3]
```diff
@@ -1,117 +1,44 @@
...
-    response.raise_for_status()
-    return response.json()['access_token']
-
 
-class LeanixAuth(requests.auth.AuthBase):
-    def __init__(self, api_token, oauth_token_url):
-        self.oauth_token_url = oauth_token_url
-        self.api_token = api_token
-        self.access_token = None
+import leanix_admin.auth as auth
```
Comment (sebastianpoeplau) [inline leanix_admin/cli.py]: You could also do `from leanix_admin import auth`...

File: leanix_admin/model.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import leanix_admin.action as action
+import leanix_admin.file as file
```
Comment (sebastianpoeplau) [inline leanix_admin/model.py]: Same remark about imports: I think it's more idiomatic to use `from leanix_admin import action`; at least in Python 2 it was 😉

File: leanix_admin/tag_group.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+            return current
+    return None
+
+
+class TagGroupsBase:
+    def __init__(self, http, graphql_url):
+        self.http = http
+        self.graphql_url = graphql_url
+
+    def _fetch_tag_groups(self, erase_id=True):
```
Comment (sebastianpoeplau) [inline leanix_admin/tag_group.py:19]: I have no idea what this function does, but it certainly looks good 😄  Would you mind adding a little doc string?

File: leanix_admin/tag_group.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+
+class TagGroupsRestoreAction(TagGroupsBase, RestoreAction):
+    def __init__(self, http, graphql_url):
+        TagGroupsBase.__init__(self, http, graphql_url)
+        RestoreAction.__init__(self, name='tag-groups')
+        self.http = http
+        self.graphql_url = graphql_url
+
+    def do_perform(self):
```
Comment (sebastianpoeplau) [inline leanix_admin/tag_group.py:85]: This is another function where I as an outsider would appreciate a little explanation...

**Comment by andrechrist** [comment]
@harti2006 I have actually no access rights to review here directly and add my recommended changes. I have added my recommended changes in a separate PR and would be happy if this can be included into master.

**Comment by harti2006** [comment]
thanks for all the feedback. I will take care of this asap, most likely next week.

**Comment by harti2006** [comment]
👍

**Comment by sebastianpoeplau** [comment]
👍

**Pull request merged**